ext/pdo_sqlite/pdo_sqlite.c

```diff
@@ -21,39 +21,654 @@
+{
+	php_stream_pdosqlite3_data *sqlite3_stream = (php_stream_pdosqlite3_data *) stream->abstract;
+
+	switch(whence) {
+		case SEEK_CUR:
+			if (offset < 0) {
+				if (sqlite3_stream->position < (size_t)(-offset)) {
+					sqlite3_stream->position = 0;
+					*newoffs = -1;
+					return -1;
+				} else {
+					sqlite3_stream->position = sqlite3_stream->position + offset;
+					*newoffs = sqlite3_stream->position;
+					stream->eof = 0;
+					return 0;
+				}
+			} else {
+				if (sqlite3_stream->position + (size_t)(offset) > sqlite3_stream->size) {
+					sqlite3_stream->position = sqlite3_stream->size;
+					*newoffs = -1;
+					return -1;
+				} else {
+					sqlite3_stream->position = sqlite3_stream->position + offset;
+					*newoffs = sqlite3_stream->position;
+					stream->eof = 0;
+					return 0;
+				}
+			}
+		case SEEK_SET:
+			if (sqlite3_stream->size < (size_t)(offset)) {
+				sqlite3_stream->position = sqlite3_stream->size;
+				*newoffs = -1;
+				return -1;
+			} else {
+				sqlite3_stream->position = offset;
+				*newoffs = sqlite3_stream->position;
+				stream->eof = 0;
+				return 0;
+			}
+		case SEEK_END:
+			if (offset > 0) {
+				sqlite3_stream->position = sqlite3_stream->size;
+				*newoffs = -1;
+				return -1;
+			} else if (sqlite3_stream->size < (size_t)(-offset)) {
+				sqlite3_stream->position = 0;
+				*newoffs = -1;
+				return -1;
+			} else {
+				sqlite3_stream->position = sqlite3_stream->size + offset;
+				*newoffs = sqlite3_stream->position;
+				stream->eof = 0;
+				return 0;
+			}
+		default:
+			*newoffs = sqlite3_stream->position;
+			return -1;
+	}
+}
+
+static int php_pdosqlite3_stream_cast(php_stream *stream, int castas, void **ret)
+{
+	return FAILURE;
+}
+
+static int php_pdosqlite3_stream_stat(php_stream *stream, php_stream_statbuf *ssb)
+{
+	php_stream_pdosqlite3_data *sqlite3_stream = (php_stream_pdosqlite3_data *) stream->abstract;
+	ssb->sb.st_size = sqlite3_stream->size;
+	return 0;
+}
+
+static const php_stream_ops php_stream_pdosqlite3_ops = {
+		php_pdosqlite3_stream_write,
+		php_pdosqlite3_stream_read,
+		php_pdosqlite3_stream_close,
+		php_pdosqlite3_stream_flush,
+		"PDOSQLite",
+		php_pdosqlite3_stream_seek,
+		php_pdosqlite3_stream_cast,
+		php_pdosqlite3_stream_stat,
+		NULL
+};
+
+/* Open a blob as a stream which we can read / write to. */
+PHP_METHOD(PdoSqlite, openBlob)
+{
+	char *table, *column, *dbname = "main", *mode = "rb";
+	size_t table_len, column_len, dbname_len;
+	zend_long rowid, flags = SQLITE_OPEN_READONLY, sqlite_flags = 0;
+	sqlite3_blob *blob = NULL;
+	php_stream_pdosqlite3_data *sqlite3_stream;
+	php_stream *stream;
+
+	pdo_dbh_t *dbh;
+	pdo_sqlite_db_handle *db_handle;
+
+	dbh = Z_PDO_DBH_P(ZEND_THIS);
+	PDO_CONSTRUCT_CHECK;
+	db_handle = (pdo_sqlite_db_handle *)dbh->driver_data;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "ssl|pl", &table, &table_len, &column, &column_len, &rowid, &dbname, &dbname_len, &flags) == FAILURE) {
```

Comment:
Should the table and column names use the ``p`` ZPP modifier?

Reply:
Hmm, yes, I'm fine with it!

Reply:
I guess the same goes for `PdoSqlite::loadExtension()`?

Reply:
Probably yes